scripts/apps/authoring/authoring/directives/AuthoringDirective.ts

```diff
@@ -542,36 +546,54 @@ export function AuthoringDirective(
              * in $scope.
              */
             $scope.publish = function() {
-                if (helpers.itemHasUnresolvedSuggestions($scope.item)) {
-                    modal.alert({
-                        headerText: gettext('Resolving suggestions'),
-                        bodyText: gettext(
-                            'Article cannot be published. Please accept or reject all suggestions first.',
-                        ),
+                $scope.$applyAsync(() => {
+                    $scope.loading = true;
+                });
+
+                return $q((resolve) => {
+                    // delay required for loading state to render
+                    // before possibly long operation (with huge articles)
+                    setTimeout(() => {
```

Comment:
if you use `$timeout` protractor will know it should wait for it, not sure if that's causing any issues but just in case